components/match2/wikis/clashofclans/match_group_input_custom.lua

```diff
@@ -460,34 +226,18 @@ function mapFunctions.readTimes(map)
 	return timesInSeconds
 end
 
--- Calculate Score and Winner of the map
-function mapFunctions.getScoresAndWinner(map)
-	map.scores = {}
-	local indexedScores = {}
-	for scoreIndex = 1, MAX_NUM_OPPONENTS do
-		-- read scores
-		local score = tonumber(map['score' .. scoreIndex]) or map['score' .. scoreIndex]
-		local obj = {}
-		if not Logic.isEmpty(score) then
-			if Logic.isNumeric(score) then
-				obj.status = 'S'
-				obj.score = tonumber(score)
-				obj.time = map.extradata.times[scoreIndex]
-				obj.percentage = map.extradata.percentages[scoreIndex] or 0
-			elseif Table.includes(ALLOWED_STATUSES, score) then
-				obj.status = score
-				obj.score = -1
-			end
-			table.insert(map.scores, score)
-			indexedScores[scoreIndex] = obj
-		else
-			break
+---@param winnerInput string|integer|nil
+---@param finished boolean
+---@return fun(opponentIndex: integer): integer?
+function MapFunctions.calculateMapScore(winnerInput, finished)
```

Comment:
Same

Reply:
same as with the other 2
mind the old score processing is bs for this wiki
it never reaches the time/percentage stuff as the draw check comes before and ignores it ...